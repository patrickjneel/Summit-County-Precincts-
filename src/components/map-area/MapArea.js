import React from 'react';
import { connect } from 'react-redux';
import './map-area.css';
import { addLocation } from '../../actions/Actions';
import MapWithAMarker from './MapWithAMarker';


const MapArea = () => { 
    return (
      <div className="map-area">
      <MapWithAMarker
        containerElement={<div style={{ height: `100%` }} />}
        mapElement={<div style={{ height: `100%` }} />}
      />
      </div>
    );
  }

export const mapStateToProps = (store) => {
  return {
    coords: store.coords 
    
  }
}

export const mapDispatchToProps = (dispatch) => {
  return {
    addLocation: (coords) => {
      dispatch(addLocation(coords))
    }
  }
}

export default connect(mapStateToProps, null)(MapArea);

