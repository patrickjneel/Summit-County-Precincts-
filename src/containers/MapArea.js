import React, { Component } from 'react';
import { connect } from 'react-redux';
import GoogleMapReact from 'google-map-react';
import './map-area.css';


class MapArea extends Component {


render () {
  return (
    <div className="map-area">
    Map goes Here
    </div>
    )
  }
}

export default MapArea;
