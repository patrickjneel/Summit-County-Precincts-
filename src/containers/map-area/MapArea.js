import React, { Component } from 'react';
import { connect } from 'react-redux';
import GoogleMapReact from 'google-map-react';
import './map-area.css';


class MapArea extends Component {
  

  render() {
    return (
      <div className="map-area">
      <GoogleMapReact
        defaultCenter={{lat: 39.5, lng: -106.06}}
        defaultZoom={12}
      />
      </div>
    );
  }
}

export default MapArea;
