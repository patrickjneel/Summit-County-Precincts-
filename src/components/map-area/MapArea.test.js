import React from 'react';
import { shallow } from 'enzyme';
import MapArea from './MapArea';

describe('MapArea Test', () => {

  it('should be defined', () => {
    const renderedMap = shallow(<MapArea />);

    expect(renderedMap).toBeDefined();
  })

  it('should match snap shot', () => {
    const renderedMap = shallow(<MapArea />);

    expect(renderedMap).toMatchSnapshot();
  })
})
