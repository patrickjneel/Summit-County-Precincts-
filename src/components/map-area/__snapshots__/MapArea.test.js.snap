// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapArea Test should match snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapArea />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <withGoogleMap(Component)
        containerElement={
                <div
                        style={
                                Object {
                                        "height": "100%",
                                      }
                        }
                />
        }
        mapElement={
                <div
                        style={
                                Object {
                                        "height": "100%",
                                      }
                        }
                />
        }
/>,
      "className": "map-area",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "containerElement": <div
          style={
                    Object {
                              "height": "100%",
                            }
          }
/>,
        "mapElement": <div
          style={
                    Object {
                              "height": "100%",
                            }
          }
/>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <withGoogleMap(Component)
          containerElement={
                    <div
                              style={
                                        Object {
                                                  "height": "100%",
                                                }
                              }
                    />
          }
          mapElement={
                    <div
                              style={
                                        Object {
                                                  "height": "100%",
                                                }
                              }
                    />
          }
/>,
        "className": "map-area",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "containerElement": <div
            style={
                        Object {
                                    "height": "100%",
                                  }
            }
/>,
          "mapElement": <div
            style={
                        Object {
                                    "height": "100%",
                                  }
            }
/>,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
